File: src/main/java/org/tud/oas/routing/IsoRaster.java
package org.tud.oas.routing;

import java.util.List;

import org.locationtech.jts.geom.Coordinate;
import org.locationtech.jts.geom.Envelope;
import org.locationtech.jts.index.kdtree.KdNode;
import org.locationtech.jts.index.kdtree.KdTree;

import com.fasterxml.jackson.annotation.JsonProperty;

public class IsoRaster {
    @JsonProperty("type")
    public String type = "Raster";

    @JsonProperty("precession")
    public double precession;

    @JsonProperty("crs")
    public String crs;

    @JsonProperty("extend")
    public double[] extend;

    @JsonProperty("size")
    public int[] size;

    @JsonProperty("envelope")
    public double[][] envelop;

    @JsonProperty("features")
    public List<GridFeature> features;

    public IsoRaster() {
    }

    private KdTree index;

    public String getType() {
        return type;
    }

    public double getPrecession() {
        return precession;
    }

    public String getCrs() {
        return crs;
    }

    public double[] getExtend() {
        return extend;
    }

    public int[] getSize() {
        return size;
    }

    public double[][] getEnvelope() {
        return this.envelop;
    }

    public void constructIndex() {
        this.index = new KdTree(500);
        for (GridFeature feature : this.features) {
            this.index.insert(new Coordinate(feature.x, feature.y), feature.value);
        }
    }

    Envelope envelope = new Envelope();

    public int getValueAtCoordinate(Coordinate coord) {
        this.envelope.init(coord.x-500, coord.x+500, coord.y-500, coord.y+500);
        List<KdNode> nodes = this.index.query(this.envelope);
        if (nodes.size() == 0) {
            return -1;
        }
        GridValue value = (GridValue)nodes.get(0).getData();
        return value.range;
    }
}

File: src/routing/IsoRaster.cs
using System;
using System.Collections.Generic;
using NetTopologySuite.Geometries;
using NetTopologySuite.Index.KdTree;

namespace DVAN.Routing 
{
    public class IsoRaster {
        public String type { get; set; }

        public double precession { get; set; }

        public String crs { get; set; }

        public double[] extend { get; set;}

        public int[] size { get; set; }

        public double[][] envelope { get; set; }

        public List<GridFeature> features { get; set; }

        public IsoRaster() {}

        private KdTree<object> index;

        public String getType() {
            return type;
        }

        public double getPrecession() {
            return precession;
        }

        public String getCrs() {
            return crs;
        }

        public double[] getExtend() {
            return extend;
        }

        public int[] getSize() {
            return size;
        }

        public double[][] getEnvelope() {
            return this.envelope;
        }

        public void constructIndex() {
            this.index = new KdTree<object>(500);
            foreach (GridFeature feature in this.features) {
                this.index.Insert(new Coordinate(feature.x, feature.y), feature.value);
            }
        }

        Envelope _envelope = new Envelope();

        public int getValueAtCoordinate(Coordinate coord) {
            this._envelope.Init(coord.X-500, coord.X+500, coord.Y-500, coord.Y+500);
            IList<KdNode<object>> nodes = this.index.Query(this._envelope);
            if (nodes.Count == 0) {
                return -1;
            }
            GridValue value = (GridValue)nodes[0].Data;
            return value.range;
        }
    }
}